### PR TITLE
BaseObject API improvements

### PR DESCRIPTION
**Pull request by mkurek**
- added serializer for each descendant model of BaseObject
- added `__str__` field to BaseObject serializer - this field provides simple way to get string representation of BaseObject (not matter what descendant type is)
- added tests to check if every descendant model of BaseObject has `__str__` field attached

### CODE REVIEW

File: src/ralph/api/utils.py [hunk 1]
```diff
@@ -69,7 +73,9 @@ def get_serializer(self, *args, **kwargs):
                     args[0].__class__
                 ]
             except KeyError:
-                pass
```
Comment (ghost) [inline src/ralph/api/utils.py:72]: On `KeyError` this `serializer_class = self.get_serializer_class()` will be used.
Is this ok?

File: src/ralph/assets/tests/test_api.py [hunk 2]
```diff
@@ -489,3 +531,21 @@ def test_tags(self):
         )
         response = self.client.get(url, format='json')
         self.assertEqual(len(response.data['results']), 1)
+
+    def test_str_field(self):
+        count = 0
+        for descendant in BaseObject._polymorphic_descendants:
+            if not descendant._polymorphic_descendants:
```
Comment (ghost) [inline src/ralph/assets/tests/test_api.py:538]: Why this check?

Comment (mkurek) [inline src/ralph/assets/tests/test_api.py:538]: There is `Asset` model, which inherits from `BaseObject`, but is not "final" type - `DataCenterAsset` and `BackOfficeAsset` are inheriting from it - so we're not exposing `assets` API endpoint and don't want to check it

Comment (ghost) [inline src/ralph/assets/tests/test_api.py:538]: :+1:

**Comment by ghost** [comment]
:ok:

File: src/ralph/api/utils.py (hunk 1 above)
Comment (mkurek) [inline src/ralph/api/utils.py:72]: imo yes. If something has no serializer we don't know how to serialize it (on dedicated view), so only base object serializer (or any other polymorphic in the future) will be used and only fields of base object will be returned.

Comment (ghost) [inline src/ralph/api/utils.py:72]: :ok:

**Comment by szok** [comment]
:ok:

**Comment by ar4s** [comment]
:ok:

**Pull request merged**